app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,51 @@
+class AddressesController < ApplicationController
+    def index 
+        address = Address.all 
+        render json: address 
+    end
+
+    def show
+      id = params[:id]
+      address = Address.find(id)
+      render json: address
+    end
+
+    def new 
+        address = Address.new 
+        render json: address
+    end
+  
+    def create
+      # param keys may subject to change depending on the body of the post request
+      address = Address.create(address_params)
+      if address.valid?
+        # message can be altered for better UI experience in the future
+        render json: {address: address, message: "Address successfully created"}
+
+      else
+        render json: {message: "There was an error."}
+      end
+    end
+  
+    def update
+      address = Address.find(params[:id])
+      if address.valid?
+        render json: {address: address, message: "Address successfully updated"}
```

Comment:
Where is the address being updated? I don't see the "update" method invoked anywhere on the address.